app/webpacker/components/NewcomerChecks/DuplicateCheckerHeader.jsx

```diff
@@ -1,34 +1,79 @@
-import React from 'react';
+import React, { useMemo } from 'react';
 import { DateTime } from 'luxon';
 import { Button, Message } from 'semantic-ui-react';
 import { duplicateCheckerJobRunStatuses } from '../../lib/wca-data.js.erb';
 
-export default function DuplicateCheckerHeader({ lastDuplicateCheckerJobRun, run }) {
-  if (
-    lastDuplicateCheckerJobRun.run_status === duplicateCheckerJobRunStatuses.in_progress
-    || lastDuplicateCheckerJobRun.run_status === duplicateCheckerJobRunStatuses.not_started
-  ) {
+// Usually this job should run in less than 5 minutes in most of the cases. It
+// can increase sometimes. A safe definition of "delay" looks like 10 minutes.
```

Comment:
This comment is essentially saying "I declare that it runs in 5 minutes, and I think that 10 minutes is reasonable".

- _Why_ is 10 minutes reasonable? You are doubling the average runtime. Why not triple it? Why not use factor 1.5?
- Does the runtime of 5 minutes scale according to number of competitors? So does "more competitors" mean "(significantly) more runtime"? Or is it more the case that the initial data fetch takes a constant 4 minutes, and all the comparisons for each competitor happen within 1 minute? This affects the decision about a good estimate.
- What makes your delay "safe"? What are the consequences of restarting a job that we believe failed (because it ran for more than your current 10 minute threshold) but actually didn't fail?

Reply:
Hm good points. Actually it depends on number of newcomers. I'm assuming that for 1 newcomer it will take around 2 seconds. If a competition has 150 newcomers, it can go up to 300 seconds (5 minutes). I feel fetching the number of newcomers here and giving a warning will be overkill, so I assumed 150 newcomers as a reasonable choice for big competitions.

What do you suggest here? IMO leaving it as 10 minutes will be better. I've modified the comments to be more clear.

Reply:
I agree that fetching the number of newcomers just to compute the threshold is overkill.
Then what about adding a visible text next to the "Reset" button, along the lines of "This button is visible because the last check did not complete within {xyz} minutes. If your competitions has a very high number of newcomers (more than 150), it may be possible that your job actually needs more time to run. Please use this button at your own discretion."